xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/jackson/modules/DidDataSerializer.java

```diff
@@ -0,0 +1,46 @@
+package org.xrpl.xrpl4j.model.jackson.modules;
+
+/*-
+ * ========================LICENSE_START=================================
+ * xrpl4j :: model
+ * %%
+ * Copyright (C) 2020 - 2022 XRPL Foundation and its contributors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdScalarSerializer;
+import org.xrpl.xrpl4j.model.transactions.DidData;
+
+import java.io.IOException;
+
+/**
+ * Custom Jackson serializer for {@link DidData}s.
+ */
+public class DidDataSerializer extends StdScalarSerializer<DidData> {
+
+  /**
+   * No-args constructor.
+   */
+  public DidDataSerializer() {
+    super(DidData.class, false);
+  }
+
+  @Override
+  public void serialize(DidData DidData, JsonGenerator gen, SerializerProvider provider) throws IOException {
+    gen.writeString(DidData.value());
```

Comment:
Should this data actually be hex-encoded? The spec shows this as a `string` in JSON, but in rippled it's just a `BLOB`.

Reply:
It should be hex encoded, but I'm not currently checking that it is in `DidData`, `DidDocument` or `DidUri`. I could do that, but it adds an extra hex decode, and I've generally stopped adding validation to our models because rippled often breaks those validations. However, if a developer sets one of these fields to a non-hex String and tries to serialize it to sign, the binary codec will throw an error because it's expecting a hex string, so their errors will at least be caught before submitting to rippled.

Reply:
Yes, that's reasonable. I'm not too worried in this particular case because the input on line 44 is going to be coming from rippled, and it _should_ be HEX.

On the constructors of these threee (`DidData.of(...)` it _might_ be worth adding validation that only HEX can be supplied there -- but to you're point, the binary codec will throw. Maybe this relates to https://github.com/XRPLF/xrpl4j/issues/500, and can be fixed there -- but the binary codec errors aren't very helpful.

This isn't a blocker for me here, so if you don't want to do anything for now, I'm good with that.